File: results/preliminary_analysis/baseline_py_results/py_source/question-27-pass-0.py
# Find the value of P and modular inverse of Q modulo 998244353. 
def calculate(p, q): 
    # Calculate the value of P and Q 
    P = p 
    Q = q 
    # Calculate the value of phi(P) 
    phi = (P - 1) * (Q - 1) 
    # Calculate the value of e 
    e = 7 
    # Calculate the value of d 
    d = modinv(e, phi) 
    return (P, Q, e, d) 
  
# Driver Code 
if __name__ == '__main__': 
    # Call the function to calculate the value of P, Q, e and d 
    (P, Q, e, d) = calculate(1000000007, 1000000009) 
    print("P = ", P) 
    print("Q = ", Q) 
    print("e = ", e) 
    print("d = ", d) 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat Mishra 
  
# This code is contributed by Rajat

File: results/preliminary_analysis/baseline_cpp_results/cpp_source/question-27-pass-0.cpp
// Find the value of P and modular inverse of Q modulo 998244353. 
 #include <bits/stdc++.h> 
 using namespace std ;
 long long calculate ( long long p , long long q ) {
 
     long long phi = ( p - 1 ) * ( q - 1 ) ;
     long long x = 1 ;
     long long y = 0 ;
     long long gcd = 1 ;
     long long temp = 0 ;
     while ( phi!= 0 ) {
         temp = phi ;
         phi = p % q ;
         x = y ;
         y = x - ( p / q ) * y ;
         gcd = temp ;
     }
     return gcd ;
 }
 
 int main ( ) {
     long long p = 998244353 ;
     long long q = 998244353 ;
     long long gcd = calculate ( p, q ) ;
     cout << "Modular Inverse of " << q << " modulo " << p << " is " << ( ( q * gcd ) % p ) << endl ;
     return 0 ;
 }
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
